lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -468,8 +455,44 @@ defmodule Screens.V2.WidgetInstance.SubwayStatus do
       |> Enum.uniq()
       |> Enum.sort()
 
-    MapSet.size(alert_trunk_stops) > 0 or
-      alert_whole_line_stops == @green_line_branches
+    alert_whole_line_stops == @green_line_branches
+  end
+
+  # If any closed stop is served by more than one branch, the alert affects the trunk
+  defp alert_affects_gl_trunk?(
+         %Alert{
+           effect: :station_closure,
+           informed_entities: informed_entities
+         },
+         gl_stop_sequences
+       ) do
+    informed_entities
+    |> Enum.map(fn e -> Map.get(e, :stop) end)
+    |> Enum.reject(&is_nil/1)
+    |> Enum.uniq()
+    |> Enum.any?(fn informed_stop ->
+      Enum.count(gl_stop_sequences, &(informed_stop in &1)) > 1
+    end)
+  end
+
+  #
+  defp alert_affects_gl_trunk?(%Alert{informed_entities: informed_entities}, gl_stop_sequences) do
+    alert_stops =
+      informed_entities
+      |> Enum.filter(fn
+        %{stop: nil} -> false
+        ie -> String.starts_with?(ie.stop, "place-") and String.starts_with?(ie.route, "Green-")
+      end)
+      |> Enum.uniq()
+
+    Enum.count(gl_stop_sequences, fn stop_sequence ->
+      alert_stops != [] and Enum.all?(alert_stops, &(&1 in stop_sequence))
+    end) > 1
```

Comment:
The result of `alert_stops != []` will be the same every time this anonymous function is called, would it make sense to do the check outside of it?
```suggestion
    alert_stops != [] and Enum.count(gl_stop_sequences, fn stop_sequence ->
      Enum.all?(alert_stops, &(&1 in stop_sequence))
    end) > 1
```

Reply:
Yes it would. Thanks!

Reply:
Or maybe for slightly better readability and performance:
```ex
# Assuming gl_stop_sequences is now a list of sets as suggested in my other comment...

alert_stops =
  # ...current pipeline, plus this extra bit
  |> MapSet.new()

if MapSet.size(alert_stops) > 0 do
  Enum.count(gl_stop_sequences, &MapSet.subset?(alert_stops, &1)) > 1
else
  false
end
```

Reply:
I feel like `MapSet`s are my most underappreciated and underutilized data structure in Elixir. This is what I should have been doing from the beginning 😅